sw/device/silicon_creator/lib/bootstrap.c

```diff
@@ -59,13 +65,47 @@ typedef enum bootstrap_state {
   kBootstrapStateProgram = 0xbdd8ca60,
 } bootstrap_state_t;
 
+/**
+ * A context object for the bootstrap state machine.
+ */
+typedef struct bootstrap_ctx {
+  bootstrap_state_t state;
+  hardened_bool_t bounds_checks_on;
+} bootstrap_ctx_t;
+
 /**
  * Handles access permissions and erases both data banks of the embedded flash.
  *
  * @return Result of the operation.
  */
 OT_WARN_UNUSED_RESULT
-static rom_error_t bootstrap_chip_erase(void) {
+static rom_error_t bootstrap_chip_erase(hardened_bool_t bounds_checks_on) {
+  if (bounds_checks_on == kHardenedBoolTrue) {
```

Comment:
I think it would be better if we could split this to a separate function per my comment above but we should use `launder32()` here:

```suggestion
  if (launder32(bounds_checks_on) == kHardenedBoolTrue) {
```

Admittedly compiler cannot deduce that this must be `kHardenedFalse` but still...

Reply:
This comment is trivially addressed now because the `bounds_check_on` branches no longer exist :)